### ISSUE TITLE
Add support for 8k tokens with Claude 3.5 Sonnet

### ISSUE THREAD

**Issue by WilliamAGH** [paul-gauthier/aider#867]
### Issue

Claude 3.5 Sonnet now supports a maximum output/response token length of 8,192 tokens (from 4,096 currently). 

To support 8k tokens, we need to add the header "anthropic-beta": "max-tokens-3-5-sonnet-2024-07-15" to our API calls:

![image](https://github.com/user-attachments/assets/d9e97ad8-0936-41a3-be73-93f92a8057be)

**Comment by flamerged** [comment]
> ### Issue
> Claude 3.5 Sonnet now supports a maximum output/response token length of 8,192 tokens (from 4,096 currently).
> 
> To support 8k tokens, we need to add the header "anthropic-beta": "max-tokens-3-5-sonnet-2024-07-15" to our API calls:
> 
> ![image](https://private-user-images.githubusercontent.com/99231285/348863225-d9e97ad8-0936-41a3-be73-93f92a8057be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExMzU4NTEsIm5iZiI6MTcyMTEzNTU1MSwicGF0aCI6Ii85OTIzMTI4NS8zNDg4NjMyMjUtZDllOTdhZDgtMDkzNi00MWEzLWJlNzMtOTNmOTJhODA1N2JlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE2VDEzMTIzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIwOWRmODg1YTg4OTc2NGIzYzlkOTM3NjEyOGU2ZDliN2IyMjY0NWY0Y2NkNTc2YjU3MTk4NDI5ZjlkYTVjMTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lw4_A_JT7XUo-u6bIZLcxU2j8w1SwS9UrjAp3_tfBiE)

How did you get past this:

<img width="580" alt="image" src="https://github.com/user-attachments/assets/18aae7b6-655f-4bef-81bf-8038fca6279a">

**Comment by pascalandy** [comment]
official docs about the feat: https://docs.anthropic.com/en/docs/about-claude/models#model-comparison

**Comment by bitnom** [comment]
did anybody submit a PR yet?

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

Aider is already able to receive unlimited output from Sonnet. See the article below. This capability was introduced back in v0.41.0.

https://aider.chat/2024/07/01/sonnet-not-lazy.html

That said, it would be beneficial to support the new 8k output limit, mainly to somewhat reduce the cost and latency of longer LLM responses.

**Comment by paul-gauthier** [comment]
I added support for 8k output tokens. 

The change is available in the main branch. You can get it by installing the latest version from github:

```
python -m pip install --upgrade git+https://github.com/paul-gauthier/aider.git
```

If you have a chance to try it, let me know if it works better for you.

**Comment by paul-gauthier** [comment]
I'm going to close this issue for now, but feel free to add a comment here and I will re-open or file a new issue any time.

**Comment by berwinsingh** [comment]
I am trying to do this on the **AWS Bedrock hosted version of 3.5 Sonnet**. However, to my surprise I found that it isn't really updated all that much meaning if I pass this parameter I get an error of extra headers not allowed. 

Then doing some searching I found that Sonnet on bedrock hasn't been updated since it was launched i.e., June 2024 which is really surprising. Any idea if Amazon or Anthropic will update the models?
